### PR TITLE
Pause menu enhancements

### PR DESCRIPTION
**Pull request by pnwparksfan**
Several improvements to `TabMissionSelectItem`:

 - Added optional dynamic scaling of mission info area based on screen size and text length in item labels
 - Added dynamic resizing of mission logo texture to support various aspect ratios
 - Refactored `Draw()` function to render correctly as a submenu item
 - Updated examples to demonstrate these new capabilities
 - Fixed texture rendering when using Rage Graphics to render outside texture files (texture alignment was previously incorrect)

Improvements to `TabSubmenuItem`:

 - Dynamic scrolling when more items are available than fit on the screen
 - Fixed bug that occurs when an interactive menu or mission select menu is focused in a submenu, and the user navigates away using LB/RB. Previously the `Focused` property of the sub menu would remain `true` but the `FocusLevel` would be out of sync, making it impossible to back out of that submenu.

### CODE REVIEW

File: Source/PauseMenu/TabMissionSelectItem.cs [hunk 1]
```diff
@@ -178,47 +280,59 @@ public override void Draw()
...
+                ResRectangle.Draw(missionPoint, new Size(logoPaddingLeft, logoArea.Height), Color.FromArgb(blackAlpha, Color.Black));
+                ResRectangle.Draw(missionPoint.AddPoints(new Point(logoArea.Width - logoPaddingRight, 0)), new Size(logoPaddingRight, logoArea.Height), Color.FromArgb(blackAlpha, Color.Black));
+            }
 
-            ResRectangle.Draw(new Point((int)res.Width - SafeSize.X - 512, SafeSize.Y + 256), new Size(512, 40), Color.FromArgb(fullAlpha, Color.Black));
-            ResText.Draw(Heists[Index].Name, new Point((int)res.Width - SafeSize.X - 4, SafeSize.Y + 260), 0.5f, Color.FromArgb(fullAlpha, Color.White), Common.EFont.HouseScript, ResText.Alignment.Right, false, false, Size.Empty);
+            // description text and background rendering
+            ResRectangle.Draw(missionPoint.AddPoints(new Point(0, logoArea.Height)), new Size(logoArea.Width, 40), Color.FromArgb(fullAlpha, Color.Black));
+            ResText.Draw(Heists[Index].Name, missionPoint.AddPoints(new Point(logoArea.Width - 4, logoArea.Height + 4)), 0.5f, Color.FromArgb(fullAlpha, Color.White), Common.EFont.HouseScript, ResText.Alignment.Right, false, false, Size.Empty);
```
Comment (alexguirre) [inline Source/PauseMenu/TabMissionSelectItem.cs:302]: Long names will look better if the background is resized and the text wrapped, like the description.

File: Source/PauseMenu/TabItem.cs [hunk 2]
```diff
@@ -31,6 +32,24 @@ public TabItem(string name)
...
+        {
+            get => Skipped;
+            set => Skipped = value;
+        }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Selected { get; set; }
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool Skipped { get; set; }
```
[the remark this answers is not in the document]
Comment (pnwparksfan) [reply Source/PauseMenu/TabItem.cs]: See aec2f36 for initial implementation of `ScrollableListBase` on `TabView` and related enhancements

**Comment by thatmaxplayle** [comment]
As per PNWParksFan's request, I'm leaving this screenshot in the discussion for this pR> 
image

**Comment by pnwparksfan** [comment]
@alexguirre do you think there should be different formatting for background color and foreground color on skipped items (in menus or tabs), like there is for disabled items on menus? And if so, should that formatting apply automatically or be optional?

File: Source/PauseMenu/TabMissionSelectItem.cs (hunk 1 above)
Comment (pnwparksfan) [reply Source/PauseMenu/TabMissionSelectItem.cs]: Resolved by d2ac036

**Comment by pnwparksfan** [comment]
> As per PNWParksFan's request, I'm leaving this screenshot in the discussion for this PR.
> 
> 1. Having some weird issue with dynamic tab widths since using a binary with these enhancements.
> 2. The text doesn't look brilliant when long strings are used. _I don't think this is specific to this PR, more of an issue in general._ But I have not tested it with the older version, so who knows... a dynamic title height (as PNW suggested) or maybe a dynamic text size for the title string?
> 
> image

#2 resolved with d3976e3

**Review by alexguirre — APPROVED** [review]

**Pull request merged**